format/cache.go

```diff
@@ -0,0 +1,93 @@
+package format
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/charmbracelet/log"
+	"github.com/numtide/treefmt/walk/cache"
+	bolt "go.etcd.io/bbolt"
+)
+
+func CompareFormatters(db *bolt.DB, formatters map[string]*Formatter) error {
+	return db.Update(func(tx *bolt.Tx) error {
+		clearPaths := false
+
+		pathsBucket, err := cache.BucketPaths(tx)
+		if err != nil {
+			return fmt.Errorf("failed to get paths bucket from cache: %w", err)
+		}
+
+		formattersBucket, err := cache.BucketFormatters(tx)
+		if err != nil {
+			return fmt.Errorf("failed to get formatters bucket from cache: %w", err)
+		}
+
+		// check for any newly configured or modified formatters
+		for name, formatter := range formatters {
+
+			stat, err := os.Lstat(formatter.Executable())
+			if err != nil {
+				return fmt.Errorf("failed to stat formatter executable %v: %w", formatter.Executable(), err)
+			}
```

Comment:
How does this interact with `--allow-missing-formatter`?

Reply:
Before this is called `cmd/format/format.go` we process the formatters like this:

```go
	for name, formatterCfg := range cfg.FormatterConfigs {
		formatter, err := format.NewFormatter(name, cfg.TreeRoot, env, formatterCfg)

		if errors.Is(err, format.ErrCommandNotFound) && cfg.AllowMissingFormatter {
			log.Debugf("formatter command not found: %v", name)
			continue
		} else if err != nil {
			return fmt.Errorf("%w: failed to initialise formatter: %v", err, name)
		}

		// store formatter by name
		formatters[name] = formatter
	}

```

If `cfg.AllowMissingFormatter` is true then the missing formatters are omitted from the formatters we will use.